packages/@ourworldindata/grapher/src/core/GrapherAnalytics.ts

```diff
@@ -47,48 +55,76 @@ export class GrapherAnalytics {
     private version: string // Ideally the Git hash commit
     private isDev: boolean
 
-    logGrapherViewError(error: Error, info: unknown): void {
-        this.logToAmplitude(EventNames.grapherViewError, { error, info })
-        this.logToGA(Categories.GrapherError, EventNames.grapherViewError)
+    logGrapherViewError(error: Error): void {
+        this.logToGA({
+            event: EventCategory.GrapherError,
+            eventAction: EventAction.grapherViewError,
+            eventContext: error.message,
+        })
     }
 
     logEntitiesNotFoundError(entities: string[]): void {
-        this.logToAmplitude(EventNames.entitiesNotFound, { entities })
-        this.logToGA(
-            Categories.GrapherError,
-            EventNames.entitiesNotFound,
-            JSON.stringify(entities)
-        )
+        this.logToGA({
+            event: EventCategory.GrapherError,
+            eventAction: EventAction.entitiesNotFound,
+            eventContext: JSON.stringify(entities),
+        })
     }
 
-    logGrapherTimelinePlay(slug?: string): void {
-        this.logToGA(Categories.GrapherUsage, EventNames.timelinePlay, slug)
+    logGlobalEntitySelector(action: entityControlEvent, note?: string): void {
+        this.logToGA({
+            event: EventCategory.GlobalEntitySelectorUsage,
+            eventAction: action,
+            eventContext: note,
+        })
     }
 
-    logGlobalEntitySelector(action: entityControlEvent, note?: string): void {
-        this.logToGA(Categories.GlobalEntitySelectorUsage, action, note)
+    logEntityPickerEvent(action: countrySelectorEvent, note?: string): void {
+        this.logToGA({
+            event: EventCategory.ExplorerCountrySelector,
+            eventAction: action,
+            eventContext: note,
+        })
     }
 
-    logEntityPickerEvent(
-        pickerSlug: string,
-        action: countrySelectorEvent,
-        note?: string
+    logGrapherClick(
+        action: string = "unknown-action",
+        label?: string,
+        grapherUrl?: string
     ): void {
-        this.logToGA(`${pickerSlug}ExplorerCountrySelectorUsage`, action, note)
+        // GA4 trims metadata fields down to 100 characters, so we want to be concise
+        const grapherUrlObj =
```

Comment:
Given the 100 character limit, you could replace "https://ourworldindata.org" with "" so that what's left is "/grapher/..."? That is, assuming some grapher slugs exceed 100 characters such that removing "https://ourworldindata.org" would allow more of the slug to be captured.

Also, does GA raise an error and fail to log the event if the metadata field is >100 characters? Just wondering if events with grapher slugs >100 chars will fail to get logged in this case.

Reply:
Ah yes, I'm actually only using `grapherUrlObj?.pathname` down below, which is exactly what you say: just `/grapher/life-expectancy` or `/explorer/migration`, for example.

But good call, I should really add a comment explaining _just that_.

Also, I might rename it to `grapherPath` or something like that, since `grapherUrl` can be misleading at this point, since it's not a full URL.

As for GA4, I checked its handling of fields >100 characters, and it just trims them down to 100 characters. There are some very few slugs that go over that limit, but that should be fine since they don't make up the majority, and are rarely embedded.